### ISSUE TITLE
DNS Service does not randomize IP addresses

### ISSUE THREAD

**Issue by narunask** [rancher/rancher#7990]
**Rancher Versions:**
```
Rancher	v1.4.1
Cattle	v0.176.9
User Interface	v1.4.6
Rancher CLI	v0.4.1
Rancher Compose	v0.12.2
```

**Docker Version:**
```
$ docker --version
Docker version 1.12.6, build 78d1802
```

**OS and where are the hosts located? (cloud, bare metal, etc):**
EC2

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node rancher, external DB

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce & Results:**
https://forums.rancher.com/t/dns-service-does-not-randomize-ip-addresses

**Comment by vincent99** [comment]
https://github.com/rancher/rancher/issues/3495

**Comment by vincent99** [comment]
This appears to have been broken when caching was added, so it's randomized once but then for the same client they get back the same cached answer.

**Comment by narunask** [comment]
Is there a chance it will be fixed in the next release?

**Comment by eesprit** [comment]
Is it fixed in 1.4.2 ?

**Comment by rene-mt** [comment]
Just for my understanding: Will there be any action taken to (re-) establish a DNS based IP randomization in Rancher? 

My company is currently evaluating Rancher as base for a large e-commerce site, based on a RancherOS-based cluster, and the current behavior is kind of a show-stopper. We want to avoid to introduce another service registry since Rancher basically has all necessary capabilities. However, "dynamic" load balancing on scaled containers with as little complexity as possible is crucial for us. For our containers we use different Linux derivates as base image (Alpine, Debian)

I saw that the Rancher DNS service accepts a "TTL" parameter, which defaults to 600 seconds (https://github.com/rancher/rancher-dns). Would it be an option to set this to zero to avoid caching respectively foster randomization of IP addresses? If so - where and how do I pass the parameter within the rancher setup?

**Comment by leodotcloud** [comment]
Fixed by `network-services: v0.2.0` and `rancher/dns:v0.15.0`.

**Comment by sangeethah** [comment]
Tested with  network-services: v0.2.0 and rancher/dns:v0.15.0 using rancher-server builds from master.

Created a service with scale 3 using image - "sangeetha/testnewhostrouting" that provides the name of the container when accessing "/name.html".

When access the sevice by its name , containers within the service are all reached randomly.

```
root@4aab247e03dc:/# curl testnewngix:80/name.html
test45-testnewngix-3
root@4aab247e03dc:/# curl testnewngix:80/name.html
test45-testnewngix-1
root@4aab247e03dc:/# curl testnewngix:80/name.html
test45-testnewngix-1
root@4aab247e03dc:/# curl testnewngix:80/name.html
test45-testnewngix-1
root@4aab247e03dc:/# curl testnewngix:80/name.html
test45-testnewngix-1
root@4aab247e03dc:/# curl testnewngix:80/name.html
test45-testnewngix-2
root@4aab247e03dc:/# curl testnewngix:80/name.html
test45-testnewngix-2
root@4aab247e03dc:/# curl testnewngix:80/name.html
test45-testnewngix-3
root@4aab247e03dc:/# curl testnewngix:80/name.html
test45-testnewngix-2
root@4aab247e03dc:/# curl testnewngix:80/name.html
test45-testnewngix-1
```

With older version of dns service - rancher/dns:v0.13.0 , DNS resolution of the service name always returned the same container's ip.
